# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1453393476 18000
#      Thu Jan 21 11:24:36 2016 -0500
# Node ID ba0a4b4f021da2e85c1da6049ddabc819c340383
# Parent  8f76684324b5561ad0c9613b6378691071512197
include errwarn.h in files that use err_disabled_feature conditionally

* file-io.cc, __delaunayn__.cc, __fltk_uigetfile__.cc,
__init_fltk__.cc, __voronoi__.cc, convhulln.cc: Include errwarn.h.
* __voronoi__.cc: Define CALLER for __err_disabled_feature when
HAVE_QHULL is not defined.

diff --git a/libinterp/corefcn/file-io.cc b/libinterp/corefcn/file-io.cc
--- a/libinterp/corefcn/file-io.cc
+++ b/libinterp/corefcn/file-io.cc
@@ -57,16 +57,17 @@ along with Octave; see the file COPYING.
 #include "error.h"
 #include "file-ops.h"
 #include "file-stat.h"
 #include "lo-ieee.h"
 #include "oct-env.h"
 #include "oct-locbuf.h"
 
 #include "defun.h"
+#include "errwarn.h"
 #include "file-io.h"
 #include "load-path.h"
 #include "oct-fstrm.h"
 #include "oct-iostrm.h"
 #include "oct-map.h"
 #include "ovl.h"
 #include "oct-prcstrm.h"
 #include "oct-stream.h"
diff --git a/libinterp/dldfcn/__delaunayn__.cc b/libinterp/dldfcn/__delaunayn__.cc
--- a/libinterp/dldfcn/__delaunayn__.cc
+++ b/libinterp/dldfcn/__delaunayn__.cc
@@ -45,16 +45,17 @@ along with Octave; see the file COPYING.
 #include <iostream>
 #include <string>
 
 #include "oct-locbuf.h"
 
 #include "Cell.h"
 #include "defun-dld.h"
 #include "error.h"
+#include "errwarn.h"
 #include "ovl.h"
 #include "unwind-prot.h"
 
 #if defined (HAVE_QHULL)
 # include "oct-qhull.h"
 # if defined (NEED_QHULL_VERSION)
 char qh_version[] = "__delaunayn__.oct 2007-08-21";
 # endif
diff --git a/libinterp/dldfcn/__fltk_uigetfile__.cc b/libinterp/dldfcn/__fltk_uigetfile__.cc
--- a/libinterp/dldfcn/__fltk_uigetfile__.cc
+++ b/libinterp/dldfcn/__fltk_uigetfile__.cc
@@ -37,16 +37,17 @@ along with Octave; see the file COPYING.
 // conflicts with Octave's Complex typedef.  We don't need the X11
 // Complex definition in this file, so remove it before including Octave
 // headers which may require Octave's Complex typedef.
 #undef Complex
 
 #endif
 
 #include "defun-dld.h"
+#include "errwarn.h"
 #include "file-ops.h"
 
 DEFUN_DLD (__fltk_uigetfile__, args, ,
            "-*- texinfo -*-\n\
 @deftypefn {} {} __fltk_uigetfile__ (@dots{})\n\
 Undocumented internal function.\n\
 @end deftypefn")
 {
diff --git a/libinterp/dldfcn/__init_fltk__.cc b/libinterp/dldfcn/__init_fltk__.cc
--- a/libinterp/dldfcn/__init_fltk__.cc
+++ b/libinterp/dldfcn/__init_fltk__.cc
@@ -35,16 +35,17 @@ To initialize:
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
 #include "builtin-defun-decls.h"
 #include "defun-dld.h"
 #include "error.h"
+#include "errwarn.h"
 #include "ov-fcn-handle.h"
 
 #ifdef HAVE_FLTK
 
 #if defined (HAVE_X_WINDOWS)
 #include <X11/Xlib.h>
 #endif
 
diff --git a/libinterp/dldfcn/__voronoi__.cc b/libinterp/dldfcn/__voronoi__.cc
--- a/libinterp/dldfcn/__voronoi__.cc
+++ b/libinterp/dldfcn/__voronoi__.cc
@@ -39,16 +39,17 @@ qhull command
 #include <list>
 
 #include "oct-locbuf.h"
 #include "lo-ieee.h"
 
 #include "Cell.h"
 #include "defun-dld.h"
 #include "error.h"
+#include "errwarn.h"
 #include "ovl.h"
 #include "unwind-prot.h"
 
 #if defined (HAVE_QHULL)
 # include "oct-qhull.h"
 # if defined (NEED_QHULL_VERSION)
 char qh_version[] = "__voronoi__.oct 2007-07-24";
 # endif
@@ -322,16 +323,21 @@ Undocumented internal function.\n\
 
   if (curlong || totlong)
     warning ("%s: qhull did not free %d bytes of long memory (%d pieces)",
              caller.c_str (), totlong, curlong);
 
   return retval;
 
 #else
+  std::string caller
+    = (args.length () > 0
+       ? args(0).xstring_value ("__voronoi__: CALLER must be a string")
+       : std::string ("__voronoi__"));
+
   err_disabled_feature (caller, "Qhull");
 #endif
 }
 
 /*
 ## No test needed for internal helper function.
 %!assert (1)
 */
diff --git a/libinterp/dldfcn/convhulln.cc b/libinterp/dldfcn/convhulln.cc
--- a/libinterp/dldfcn/convhulln.cc
+++ b/libinterp/dldfcn/convhulln.cc
@@ -33,16 +33,17 @@ 2006-05-01 Tom Holroyd
 #include <config.h>
 #endif
 
 #include "oct-locbuf.h"
 
 #include "Cell.h"
 #include "defun-dld.h"
 #include "error.h"
+#include "errwarn.h"
 #include "ovl.h"
 #include "parse.h"
 #include "unwind-prot.h"
 
 #if defined (HAVE_QHULL)
 # include "oct-qhull.h"
 # if defined (NEED_QHULL_VERSION)
 char qh_version[] = "convhulln.oct 2007-07-24";
